### PR TITLE
RA-37 Prevent double contacts 

### PR DESCRIPTION
**Pull request by trondsevre**
By preventing internal contacts, parameter based

### CODE REVIEW

File: src/main/java/no/fint/sikri/data/noark/korrespondansepart/KorrespondansepartFactory.java [hunk 1]
```diff
@@ -18,6 +19,10 @@
 
 @Service
 public class KorrespondansepartFactory {
+
+    @Value("${fint.sikri.droppinternekorrespondanseparter:false}")
```
Comment (asgeirn) [inline src/main/java/no/fint/sikri/data/noark/korrespondansepart/KorrespondansepartFactory.java:23]: Convention for property naming is English, and chaining words with hyphens.

I'd suggest `fint.sikri.skip-internal-contacts`.

File: src/main/java/no/fint/sikri/data/noark/korrespondansepart/KorrespondansepartFactory.java [hunk 2]
```diff
@@ -101,7 +106,7 @@ public SenderRecipientType createSenderRecipient(KorrespondansepartResource inpu
                 .map(Link::getHref)
                 .map(s -> StringUtils.substringAfterLast(s, "/"))
                 .map(StringUtils::upperCase)
-                .filter(s -> StringUtils.equalsAny(s, "EA", "EM", "EK", "IA", "IM", "IK"))
+                .filter(s -> StringUtils.equalsAny(s, "EA", "EM", "EK", "IA", "IM", "IK") && (!skipIntern || !StringUtils.equalsAny(s, "IA", "IM", "IK")))
```
Comment (asgeirn) [inline src/main/java/no/fint/sikri/data/noark/korrespondansepart/KorrespondansepartFactory.java:109]: For clarity, consider using the parameter from above to initialize the matching list:

```java
String[] expectedContactTypes = 
    skipInternalContacts ?
    new String[] { "EA", "EM", "EK" } :
    new String[] { "EA", "EM", "EK", "IA", "IM", "IK" };
```

**Review by asgeirn — CHANGES_REQUESTED** [review]

File: src/main/java/no/fint/sikri/data/noark/korrespondansepart/KorrespondansepartFactory.java (hunk 1 above)
Comment (trondsevre) [reply src/main/java/no/fint/sikri/data/noark/korrespondansepart/KorrespondansepartFactory.java]: Updated

File: src/main/java/no/fint/sikri/data/noark/korrespondansepart/KorrespondansepartFactory.java (hunk 2 above)
Comment (trondsevre) [reply src/main/java/no/fint/sikri/data/noark/korrespondansepart/KorrespondansepartFactory.java]: Updated

**Review by asgeirn — APPROVED** [review]
Much better!

**Pull request merged**